contracts/registries/LicenseRegistryV2.sol

```diff
@@ -0,0 +1,382 @@
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        if (index >= $.attachedLicenseConfigs[ipId].length()) {
+            revert Errors.LicenseRegistry__IndexOutOfBounds(ipId, index);
+        }
+        licenseTemplate = $.licenseTemplates[ipId];
+        licenseConfigId = $.attachedLicenseConfigs[ipId].at(index);
+    }
+
+    function getAttachedLicenseConfigCount(address ipId) external view returns (uint256) {
+        return _getLicenseRegistryStorage().attachedLicenseConfigs[ipId].length();
+    }
+
+    function getMintingLicenseSpec(
+        address ipId,
+        address licenseTemplate,
+        uint256 licenseConfigId
+    ) external view returns (Licensing.MintingLicenseSpec memory) {
+        return _getMintingLicenseSpec(ipId, licenseTemplate, licenseConfigId);
+    }
+
+    /// @notice Returns the canonical protocol-wide LicensingModule
+    function licensingModule() external view returns (ILicensingModule) {
+        return _getLicenseRegistryStorage().licensingModule;
+    }
+
+    /// @notice Returns the canonical protocol-wide DisputeModule
+    function disputeModule() external view returns (IDisputeModule) {
+        return _getLicenseRegistryStorage().disputeModule;
+    }
+
+    function getExpireTime(address ipId) external view returns (uint256) {
+        return _getLicenseRegistryStorage().expireTimes[ipId];
+    }
+
+    function getDefaultLicenseConfig() external view returns (address licenseTemplate, uint256 licenseConfigId) {
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        return ($.defaultLicenseTemplate, $.defaultLicenseConfigId);
+    }
+
+    function _setExpireTime(address ipId, uint256 expireTime) internal {
+        _getLicenseRegistryStorage().expireTimes[ipId] = expireTime;
+        emit ExpireTimeSet(ipId, expireTime);
+    }
+
+    function _isDerivativeIp(address derivativeIpId) internal view returns (bool) {
+        return _getLicenseRegistryStorage().originalIps[derivativeIpId].length() > 0;
+    }
+
+    function _getMintingLicenseSpec(
+        address ipId,
+        address licenseTemplate,
+        uint256 licenseConfigId
+    ) internal view returns (Licensing.MintingLicenseSpec memory) {
+        LicenseRegistryStorage storage $ = _getLicenseRegistryStorage();
+        if (!$.registeredLicenseTemplates[licenseTemplate]) {
+            revert Errors.LicenseRegistry__UnregisteredLicenseTemplate(licenseTemplate);
+        }
+        if ($.mintingLicenseSpecs[_getHash(ipId, licenseTemplate, licenseConfigId)].isSet) {
+            return $.mintingLicenseSpecs[_getHash(ipId, licenseTemplate, licenseConfigId)];
+        }
+        return $.mintingLicenseSpecsForAll[ipId];
+    }
+
+    function _getHash(address ipId, address licenseTemplate, uint256 licenseConfigId) internal pure returns (bytes32) {
+        return keccak256(abi.encodePacked(ipId, licenseTemplate, licenseConfigId));
```

Comment:
`abi.encodePacked` could theoretically result in different inputs generating the same hash due to packing.
`keccak256` reduces its input to bytes32, so there is no difference in storage savings.
Please use `abi.encode`

Reply:
Also, `_getHash` is too generic of a name, more descriptive names would be better

Reply:
Agreed! updated and renamed the function.